newIDE/app/src/MainFrame/index.js

```diff
@@ -2180,6 +2183,28 @@ const MainFrame = (props: Props) => {
     return true;
   };
 
+  const canInstallPrivateAsset = async (): Promise<boolean> => {
+    const storageProvider = getStorageProvider();
+    // A private asset can always be installed locally, as it will be downloaded.
+    if (storageProvider.internalName === 'LocalFile') {
+      return true;
+    }
+    // A private asset can be installed on the cloud if the user has saved their project as a cloud project.
+    if (storageProvider.internalName === 'Cloud') {
+      if (!currentFileMetadata) {
+        return false;
+      }
+      const cloudProjectId = currentFileMetadata.fileIdentifier;
+
+      const cloudProject = await getCloudProject(
+        authenticatedUser,
+        cloudProjectId
+      );
+      return !!cloudProject;
```

Comment:
You should not need to do this. If the storageProvider is Cloud, the file was either loaded from the Cloud or saved as a cloud project. No need to do a `getCloudProject` request every time we need to know if we can install a private asset.

Reply:
In other words, the storageProvider.internalName is the name of the storage provider for the SAVED project. This should be a guarantee. 
- On the desktop app, it's a LocalFile (or soon, a Cloud) storage provider. The file is already saved locally, or will be created in the Cloud.
- On the web app, it's either a UrlStorageProvider (in which case the file is "saved" in a url, so almost nothing is supported) or Cloud (in which case it's saved online).

Reply:
Actually, there is one case where it's useful.
When you:
- arrive on the webapp (not storage provider)
- create a project
- save it in the cloud (storage provider = cloud)
- create a new project (storage provider still = cloud)
-> we're then in the case where the new project is not saved yet, and yet the default storage provider is cloud.

If what you're saying is true, then we should be clearing the storage provider when opening up a new project, and it doesn't seem to be the case

Reply:
> If what you're saying is true, then we should be clearing the storage provider when opening up a new project, and it doesn't seem to be the case

Yes. We should reset it to null or UrlStorageProvider, let's see that on another PR and keep a note here in this PR description :)

Reply:
I've made it here, as it was quite easy to implement. I'll let you review the strategy I picked